src/client/dfuse/il/int_read.c

```diff
@@ -72,29 +66,47 @@ read_bulk(char *buff, size_t len, off_t position, struct fd_entry *entry, int *e
 ssize_t
 ioil_do_pread(char *buff, size_t len, off_t position, struct fd_entry *entry, int *errcode)
 {
-	return read_bulk(buff, len, position, entry, errcode);
+	d_iov_t     iov = {};
+	d_sg_list_t sgl = {};
+
+	sgl.sg_nr = 1;
+	d_iov_set(&iov, (void *)buff, len);
+	sgl.sg_iovs = &iov;
+
+	return read_bulksgl(&sgl, len, position, entry, errcode);
 }
 
 ssize_t
 ioil_do_preadv(const struct iovec *iov, int count, off_t position, struct fd_entry *entry,
 	       int *errcode)
 {
-	ssize_t bytes_read;
-	ssize_t total_read = 0;
-	int     i;
+	d_iov_t    *diov;
+	d_sg_list_t sgl        = {};
+	ssize_t     total_read = 0;
+	int         i;
+	int         rc;
+	int         new_count;
+
+	D_ALLOC_ARRAY(diov, count);
+	if (diov == NULL) {
+		*errcode = ENOMEM;
+		return -1;
+	}
 
-	for (i = 0; i < count; i++) {
-		bytes_read = read_bulk(iov[i].iov_base, iov[i].iov_len, position, entry, errcode);
+	for (i = 0, new_count = 0; i < count; i++) {
+		/** See DAOS-15089. This is a workaround */
+		if (iov[i].iov_len == 0)
+			continue;
```

Comment:
I agree that this should be addressed/supported at the lower layer but this approach is OK for getting this feature landed.

Reply:
Hmm, I ran with 

Features: dfuse

Is there something wrong with handling or am I doing this wrong?

Reply:
So you did,  I was looking at an older commit.